src/components/Map.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+import { InfoWindow, Marker } from "google-maps-react";
+import MapContainer from "../container/MapContainer";
+import InfoComponent from "../components/InfoComponent";
+import pin from "../assets/images/pin.png";
+
+export class Map extends Component {
+  state = {
+    showingInfoWindow: false,
+    activeMarker: {},
+    selectedPlace: {},
+    list: []
+  };
+
+  onMarkerClick = (props, marker, e) => {
+    this.setState({
+      selectedPlace: props,
+      activeMarker: marker,
+      showingInfoWindow: true
+    });
+  };
+
+  onClose = () => {
+    if (this.state.showingInfoWindow) {
+      this.setState({
+        showingInfoWindow: false,
+        activeMarker: null
+      });
+    }
+  };
+  setLocation = list => {
+    this.setState({ list: list });
+    this.props.setLocations(list);
+  };
+  render() {
+    return (
+      <MapContainer
+        centerAroundCurrentLocation={this.props.centerAroundCurrentLocation}
+        google={this.props.google}
+        setLocations={list => {
+          this.setLocation(list);
+        }}
+        selectedPlace={this.props.selectedPlace}
+      >
+        {this.state.list && this.state.list.length > 0
```

Comment:
Use destructuring wherever possible. In this file, there are enough places where we can replace it.